internal/extproc/translator/jsonschema_helper.go

```diff
@@ -0,0 +1,408 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package translator
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"google.golang.org/genai"
+)
+
+// get the supported field names
+func jsonSchemaGetAllowedFields() map[string]struct{} {
+	fieldSet := make(map[string]struct{})
+	var schema *genai.Schema
+	t := reflect.TypeOf(schema).Elem()
```

Comment:
i think the list of allowed fields is static, can we hardcode it instead
wanna avoid reflection.
field list: https://cloud.google.com/vertex-ai/generative-ai/docs/reference/rest/v1/Schema

also, i dont expect the list to change that often